atlp-rwanda/Phantom-Backend-Pirates
# Sign in to phantom

#### Main object of the Task
The driver and operator should be able to login ith credentials provided by the admin, so that they can perform their tasks. 
#### Description of Task to be completed
1. The driver, admin and operator should be able to sign in to the app by providing their email & password
2. The API should return a JW toke on a successful login
3. The administror should be able to sign in to the app using pre-configured login credentials
4. Appropriate error messages are displayed on unsuccessful attempts
#### How should this be manually tested
1. First and foremost clone the repo, cd into it and `run npm run seed` to help you create the dummy data in the seeder folder
2. You can run tests by running `npm run test` in your terminal
3. Run the server by running `npm run dev` in your terminal
4. Using Postman test the POST: login endpoint (`http://localhost:3000/users/login`) which by sending the email and password in the user_dummy data. 
_key:_Note that the postman body must be raw and json content. After a successful login you will see that a token will be generated.
#### Screenshots 
![Successful Login](https://user-images.githubusercontent.com/16195731/156445411-dfe30bf5-08dd-4589-9f04-ec6d08a42df3.jpg)
![Unsuccessful login](https://user-images.githubusercontent.com/16195731/156445405-389ccad3-acf9-4d07-be8a-c1796194b212.jpg)
![User's email does exist](https://user-images.githubusercontent.com/16195731/156445392-2cd1e9c8-f457-4434-aa53-23efae62a3e1.jpg)
